integration_tests/juniper_tests/src/codegen/proc_macro_param_attrs.rs

```diff
@@ -0,0 +1,211 @@
+static SCHEMA_INTROSPECTION_QUERY: &str = r#"
+    query IntrospectionQuery {
+      __schema {
+        types {
+          ...FullType
+        }
+      }
+    }
+
+    fragment FullType on __Type {
+      name
+      fields(includeDeprecated: true) {
+        name
+        description
+        args {
+          ...InputValue
+        }
+      }
+    }
+
+    fragment InputValue on __InputValue {
+      name
+      description
+      defaultValue
+    }
+"#;
+
+#[test]
+fn old_descriptions_applied_correctly() {
+    let schema = introspect_schema();
+    let query = schema.types.iter().find(|ty| ty.name == "Query").unwrap();
+
+    let field = query
+        .fields
+        .iter()
+        .find(|field| field.name == "fieldOldAttrs")
+        .unwrap();
+
+    let arg1 = field.args.iter().find(|arg| arg.name == "arg1").unwrap();
+    assert_eq!(&arg1.description, &Some("arg1 desc".to_string()));
+    assert_eq!(
+        &arg1.default_value,
+        &Some(Value::String("true".to_string()))
+    );
+
+    let arg2 = field.args.iter().find(|arg| arg.name == "arg2").unwrap();
+    assert_eq!(&arg2.description, &Some("arg2 desc".to_string()));
+    assert_eq!(
+        &arg2.default_value,
+        &Some(Value::String("false".to_string()))
+    );
+}
+
+#[test]
+fn new_descriptions_applied_correctly() {
+    let schema = introspect_schema();
+    let query = schema.types.iter().find(|ty| ty.name == "Query").unwrap();
+
+    let field = query
+        .fields
+        .iter()
+        .find(|field| field.name == "fieldNewAttrs")
+        .unwrap();
+
+    let arg1 = field.args.iter().find(|arg| arg.name == "arg1").unwrap();
+    assert_eq!(&arg1.description, &Some("arg1 desc".to_string()));
+    assert_eq!(
+        &arg1.default_value,
+        &Some(Value::String("true".to_string()))
+    );
+
+    let arg2 = field.args.iter().find(|arg| arg.name == "arg2").unwrap();
+    assert_eq!(&arg2.description, &Some("arg2 desc".to_string()));
+    assert_eq!(
+        &arg2.default_value,
+        &Some(Value::String("false".to_string()))
+    );
+}
+
+#[derive(Debug)]
+struct Schema {
+    types: Vec<Type>,
+}
+
+#[derive(Debug)]
+struct Type {
+    name: String,
+    fields: Vec<Field>,
+}
+
+#[derive(Debug)]
+struct Field {
+    name: String,
+    args: Vec<Arg>,
+    description: Option<String>,
+}
+
+#[derive(Debug)]
+struct Arg {
+    name: String,
+    description: Option<String>,
+    default_value: Option<Value>,
+}
+
+fn introspect_schema() -> Schema {
+    let (value, _errors) = juniper::execute(
+        SCHEMA_INTROSPECTION_QUERY,
+        None,
+        &RootNode::new(Query, juniper::EmptyMutation::new()),
+        &Variables::new(),
+        &(),
+    )
+    .unwrap();
+
+    let value: Value = serde_json::from_str(&serde_json::to_string(&value).unwrap()).unwrap();
+
+    let types = value["__schema"]["types"]
+        .as_array()
+        .unwrap()
+        .iter()
+        .map(parse_type)
```

Comment:
This manual parsing is tedious, but I wasn't able to get serde derives working 🤔

Reply:
This file has been removed since the old style is no longer supported.